warning: implicit capture of ‘this’ via ‘[=]’ is deprecated in C++20

<pre>../../Simplex_tree/include/gudhi/Simplex_tree.h: In lambda function:
../../Simplex_tree/include/gudhi/Simplex_tree.h:1381:58: warning: implicit capture of ‘this’ via ‘[=]’ is deprecated in C++20 [-Wdeprecated]
 1381 |     auto last = std::remove_if(list.begin(), list.end(), [=](Dit_value_t& simplex) {
      |                                                          ^
../../Simplex_tree/include/gudhi/Simplex_tree.h:1381:58: note: add explicit ‘this’ or ‘*this’ capture
</pre>

--- a/src/Nerve_GIC/include/gudhi/GIC.h
+++ b/src/Nerve_GIC/include/gudhi/GIC.h
@@ -707,7 +707,7 @@ class Cover_complex {
     // Sort points according to function values
     std::vector<int> points(n);
     for (int i = 0; i < n; i++) points[i] = i;
-    std::sort(points.begin(), points.end(), [=](const int & p1, const int & p2){return (this->func[p1] < this->func[p2]);});
+    std::sort(points.begin(), points.end(), [this](int p1, int p2){return (this->func[p1] < this->func[p2]);});
 
     int id = 0;
     int pos = 0;

--- a/src/Simplex_tree/include/gudhi/Simplex_tree.h
+++ b/src/Simplex_tree/include/gudhi/Simplex_tree.h
@@ -1378,7 +1378,7 @@ class Simplex_tree {
  private:
   bool rec_prune_above_filtration(Siblings* sib, Filtration_value filt) {
     auto&& list = sib->members();
-    auto last = std::remove_if(list.begin(), list.end(), [=](Dit_value_t& simplex) {
+    auto last = std::remove_if(list.begin(), list.end(), [this,filt](Dit_value_t& simplex) {
         if (simplex.second.filtration() <= filt) return false;
         if (has_children(&simplex)) rec_delete(simplex.second.children());
         // dimension may need to be lowered

